velocity-engine-core/src/main/java/org/apache/velocity/util/ArrayIterator.java

```diff
@@ -42,7 +37,8 @@
  * @author <a href="mailto:geirm@apache.org">Geir Magnusson Jr.</a>
  * @version $Id$
  */
-public class ArrayIterator implements Iterator
+public class ArrayIterator
+    implements Iterator<Object>
```

Comment:
Can this be parameterized instead?

    public class ArrayIterator<T> implements Iterator<T>

After a quick glance at the code, this seems impossible to do correctly, but maybe you have more ideas.